languages/ruby/exercises/concept/booleans/.docs/introduction.md

```diff
@@ -0,0 +1,60 @@
+## True and False
+
+True and false logical states are represented with `true` and `false` in Ruby. These may either be used as literals on their own, or as a result of logical or comparison methods.
+
+```ruby
+true_variable = true
+false_variable = false
+
+true && false
+# => false
+
+1 < 2
+# => true
+```
+
+## _Truthy_ and _falsey_
+
+When evaluating objects in `if` statements or other boolean contexts, all objects evaluate as _truthy_ **except** for `false` and `nil`.
+
+## Control flow
+
+_Truthy_ and _falsey_ evaluations are useful in the context of control flow. Like in procedural languages, Ruby has an `if`...`else` construct, but it may be more common to use `if` as a "guarding" statement to modify the evaluation of an expression.
+
+```ruby
+def falsey
+  nil || false
+end
+
+def truthy
+  not falsey
+end
+
+if truthy
+  # this block is evaluated
+end
+
+if falsey
+  # this block is not evaluated
+else
+  # this block is evaluated
+end
+
+1 + 1 if truthy
+# => this will evaluate and return 2
+
+2 + 2 if falsey
+# => the numbers are not added because of the modifier, nil is returned
+```
+
+If it is preferred to use the positive form of a conditional statement. Rather than negating a negative statement, `unless` is provided as an alternative. E.g. `eat_desert unless too_full`
```

Comment:
The first sentence doesn't quite make sense.

Reply:
Maybe it would be better with an example.

```suggestion
Ruby provides `unless` so that it isn't required to have double negative statements.  E.g. Rather than `eat_desert if not too_full`, it would be better to write `eat_desert unless too_full`
```

Reply:
It's because the first sentence is not supposed to be a sentence, but a comma, and a lower case "r" for "rather" and appears to be an adulteration of the original thing written.

> If it is preferred to use the positive form of a conditional statement, rather than negating a negative statement, we can use positive conditional statements always and use `if` and `unless` to communicate the positive or negative overall statement. -- _paraphrased, not quoted_.

Reply:
Thoughts on the suggested change?

Reply:
I would leave the double negation mention out. as that acts on the conditional statement but is not a change in "boolean" terms, just a strict change from "boolean" to "Boolean".

```suggestion
If it is preferred to use always use a positive conditional statement, we can use `unless` to negate the if statement itself.

 E.g.:
    eat_desert unless too_full

 compared to

    eat_desert if not too_full
```

Reply:
"It may be that we prefer to always use"

Reply:
```suggestion
Ruby also provides `unless` to make code read well, i.e., rather than `eat_desert if not too_full`, we can also write `eat_desert unless too_full`
```

Reply:
This one… (though "also" can come out.